broadinstitute/single_cell_portal_core
# Updating to latest release of tcell_agent (SCP-4576)

#### BACKGROUND & CHANGES
Updates to the most recent version of `tcell_agent` to address the following error when attempting to write to the visualization API cache when TCell is enabled:
```
{
  "error": "no implicit conversion of Hash into Integer",
  "error_class": "TypeError",
  "source": "/usr/local/rvm/gems/ruby-3.1.2/gems/tcell_agent-2.3.0/lib/tcell_agent/instrumentation/monkey_patches/file.rb:22:in `initialize'"
}
```
#### MANUAL TESTING
1. Pull this branch
2. Open `Gemfile` and add `gem 'tcell_agent'` after `gem 'net-pop'`
3. Run `bundle install`
4. From the project root, run `touch tmp/caching-dev.txt` to ensure caching is turned on
5. Start server as normal, then navigate to any study with visualizations turned on
6. Click around in the Explore tab, confirming that visualization load
7. In `development.log`, note that values are being written/read to the cache with messages like the following:
```
Writing to API cache: _single_cell_api_v1_studies_SCP13_explore_
...
Reading from API cache: _single_cell_api_v1_studies_SCP13_clusters__default_cluster_name__default_fields_coordinates_cells_annotation
Filter chain halted as :check_api_cache! rendered or redirected
...

```
